`SpectralCoord` values cause validation failure in `astropy.modeling.bounding_box._Interval`

<!-- This comments are hidden when you submit the issue,
so you do not need to remove them! -->

<!-- Please be sure to check out our contributing guidelines,
https://github.com/astropy/astropy/blob/main/CONTRIBUTING.md .
Please be sure to check out our code of conduct,
https://github.com/astropy/astropy/blob/main/CODE_OF_CONDUCT.md . -->

<!-- Please have a search on our GitHub repository to see if a similar
issue has already been posted.
If a similar issue is closed, have a quick look to see if you are satisfied
by the resolution.
If not please go ahead and open an issue! -->

<!-- Please check that the development version still produces the same bug.
You can install development version with
pip install git+https://github.com/astropy/astropy
command. -->

### Description
Issue astropy/specutils#893 some of the issues described by @rosteen indicate that In astropy 5.0 there can be some issues when bounding boxes validate their interval components. In particular, these errors seem to occur when interval bounds are assigned as `SpectralCoord` types.

The following should reproduce the error in a more controlled fashion:
```python
import astropy.units as u
from astropy.modeling.bounding_box import _Interval
from astropy.coordinates import SpectralCoord

lower = SpectralCoord(1, u.um)
upper = SpectralCoord(10, u.um)
interval = _Interval.validate((lower, upper))
```

This interval should validate and be constructed without issue. However, it currently produces the following traceback:
```python
interval = (<SpectralCoord 1. um>, <SpectralCoord 10. um>)

    @staticmethod
    def _validate_shape(interval):
        """Validate the shape of an interval representation"""
        MESSAGE = """An interval must be some sort of sequence of length 2"""

        try:
            shape = np.shape(interval)
        except TypeError:
            try:
                # np.shape does not work with lists of Quantities
                if len(interval) == 1:
                    interval = interval[0]
                shape = np.shape([b.to_value() for b in interval])
            except (ValueError, TypeError, AttributeError):
>               raise ValueError(MESSAGE)
E               ValueError: An interval must be some sort of sequence of length 2

bounding_box.py:73: ValueError
```

--- a/astropy/coordinates/spectral_quantity.py
+++ b/astropy/coordinates/spectral_quantity.py
@@ -297,5 +297,8 @@ class SpectralQuantity(SpecificTypeQuantity):
 
         return result
 
-    def to_value(self, *args, **kwargs):
-        return self.to(*args, **kwargs).value
+    def to_value(self, unit=None, *args, **kwargs):
+        if unit is None:
+            return self.view(np.ndarray)
+
+        return self.to(unit, *args, **kwargs).value

--- a/astropy/modeling/tests/test_bounding_box.py
+++ b/astropy/modeling/tests/test_bounding_box.py
@@ -10,6 +10,7 @@ from astropy.modeling.bounding_box import (_BaseInterval, _Interval, _ignored_in
                                            CompoundBoundingBox)
 from astropy.modeling.models import Gaussian1D, Gaussian2D, Shift, Scale, Identity
 from astropy.modeling.core import Model, fix_inputs
+from astropy.coordinates import SpectralCoord
 import astropy.units as u
 
 
@@ -187,6 +188,16 @@ class Test_Interval:
             assert num < _ignored_interval[1]
 
             assert not (_ignored_interval.outside(np.array([num]))).all()
+
+    def test_validate_with_SpectralCoord(self):
+        """Regression test for issue #12439"""
+
+        lower = SpectralCoord(1, u.um)
+        upper = SpectralCoord(10, u.um)
+
+        interval = _Interval.validate((lower, upper))
+        assert interval.lower == lower
+        assert interval.upper == upper
 
 
 class Test_BoundingDomain:

--- a/docs/changes/coordinates/12440.api.rst
+++ b/docs/changes/coordinates/12440.api.rst
@@ -1,0 +1,2 @@
+``SpectralQuantity`` and ``SpectralCoord`` ``.to_value`` method can now be called without
+``unit`` argument in order to maintain a consistent interface with ``Quantity.to_value``

